WordPress/gutenberg
# Refactor/toolbar item component to typescript

## What?

Convert the ToolbarItem component to TypeScript.

## Why?

This PR is needed as part of an effort to convert ToolbarItem to TypeScript. As mentioned in #35744 .

## How?

By adding types to ToolbarItem.

## Testing Instructions




Create or otherwise open a post
Add a block that would have some toolbar controls (e.g. Image)
Expected: Toolbar buttons should fundamentally work the same as before:
